src/zenml/stack_stores/base_stack_store.py

```diff
@@ -293,7 +316,48 @@ def __check_component(
             typ: name for typ, name in map(__check_component, stack.components)
         }
         metadata = {c.type.value: c.flavor for c in stack.components}
-        self._create_stack(stack.name, stack_configuration)
+        self._save_stack(stack.name, stack_configuration)
+        return metadata
+
+    def update_stack(self, stack: StackWrapper) -> Dict[str, str]:
+        """Update a stack and its components.
+
+        If any of the stack's components aren't registered in the stack store
+        yet, this method will try to register them as well.
+
+        Args:
+            stack: The stack to update.
+
+        Returns:
+            metadata dict for telemetry or logging.
+
+        Raises:
+            DoesNotExistException: If no stack exists with the given name.
+        """
+        try:
+            self.get_stack(stack.name)
+        except KeyError:
+            raise DoesNotExistException(
+                f"Unable to update stack with name '{stack.name}': No existing "
+                f"stack found with this name."
+            )
+
+        def __check_component(
+            component: StackComponentWrapper,
+        ) -> Tuple[StackComponentType, str]:
+            try:
+                _ = self.get_stack_component(
+                    component_type=component.type, name=component.name
+                )
+            except KeyError:
+                self.register_stack_component(component)
+            return component.type, component.name
+
+        stack_configuration = {
+            typ: name for typ, name in map(__check_component, stack.components)
+        }
```

Comment:
Maybe we can extract this common code from both registering and updating a stack into some separate function that both of them can call?